test/entities/swapV2.integration.test.ts

```diff
@@ -0,0 +1,210 @@
+            },
+        ],
+        pools: [
+            '0x5c6ee304399dbdb9c8ef030ab642b10820db8f56000200000000000000000014',
+        ],
+        inputAmountRaw: 100000000000n,
+        outputAmountRaw: 100000000000n,
+    };
+
+    beforeAll(async () => {
+        client = createTestClient({
+            mode: 'anvil',
+            chain: CHAINS[chainId],
+            transport: http(rpcUrl),
+        })
+            .extend(publicActions)
+            .extend(walletActions);
+
+        testAddress = (await client.getAddresses())[0];
+    });
+
+    beforeEach(async () => {
+        await forkSetup(
+            client,
+            testAddress,
+            [WETH.address, BAL.address],
+            [WETH.slot as number, BAL.slot as number],
+            [parseEther('100'), parseEther('100')],
+            undefined,
+            balancerVersion,
+        );
+    });
+
+    describe('query method should return correct updated', () => {
+        test('GivenIn', async () => {
+            const swap = new SwapV2({
+                chainId,
+                paths: [pathBalWeth],
+                swapKind: SwapKind.GivenIn,
+            });
+
+            const updated = await swap.query(rpcUrl);
+
+            const wethToken = new Token(
+                chainId,
+                TOKENS[chainId].WETH.address,
+                TOKENS[chainId].WETH.decimals,
+            );
+            expect(updated.token).to.deep.eq(wethToken);
+            expect(updated.amount).to.eq(44236888n);
+        });
+        test('GivenOut', async () => {
+            const swap = new SwapV2({
+                chainId,
+                paths: [pathBalWeth],
+                swapKind: SwapKind.GivenOut,
+            });
+
+            const updated = await swap.query(rpcUrl);
+
+            const balToken = new Token(
+                chainId,
+                TOKENS[chainId].BAL.address,
+                TOKENS[chainId].BAL.decimals,
+            );
+            expect(updated.token).to.deep.eq(balToken);
+            expect(updated.amount).to.eq(60635225778147n);
+        });
+    });
+    describe('swap should be executed correcly', () => {
+        describe('wethIsEth: false', () => {
+            const swapParams = {
+                chainId,
+                paths: [pathBalWeth],
+                wethIsEth: false,
+            };
+            test('GivenIn', async () => {
+                const swap = new Swap({
+                    ...swapParams,
+                    swapKind: SwapKind.GivenIn,
+                });
+                await assertSwapExactIn(client, rpcUrl, chainId, swap, false);
+            });
+            test('GivenOut', async () => {
+                const swap = new Swap({
+                    ...swapParams,
+                    swapKind: SwapKind.GivenOut,
+                });
+                await assertSwapExactOut(client, rpcUrl, chainId, swap, false);
+            });
+        });
+        describe('wethIsEth: true', () => {
```

Comment:
It took me a while to identify that wethIsEth is set through an input at assertSwapExactIn 😅 
And only then I realized that `doSwap` is actually happening inside `assertSwap`.
I'd say that it's ok to leave as is for now, but it would be nice to have a small low priority task to refactor this later on.
Let me know and I'll create one so we don't forget to get back to it.

Reply:
Did a small refactor that should make wethIsEth a bit clearer.